[BOUNTY] Code Review Bounty Program — Review PRs, Earn RTC (100 RTC Pool)

## Code Review Bounty Program

**Pool: 200 RTC (ongoing)**

### Reward Structure
| Review Quality | Reward per PR |
|---------------|---------------|
| **Thorough** — Line-by-line with actionable feedback | 10-15 RTC |
| **Standard** — Functional review with useful comments | 5-10 RTC |
| **Security-focused** — Identifies vulnerabilities | 15-25 RTC |

### Eligible Repositories
- Scottcjn/Rustchain, Scottcjn/bottube, Scottcjn/rustchain-bounties, Scottcjn/ram-coffers

### How to Claim
1. Review an open PR with detailed comments
2. Comment on this issue linking your review
3. Maintainer assesses quality and assigns reward

### Bonus
- Find a bug that would have shipped: +10 RTC
- Find a security vuln in a PR: +25 RTC

---
*Total RTC supply: 8.3M | RTC Reference Rate: 1 RTC = $0.10 USD*

## RustChain review bounty claim update — cumulative 285 reviews

Reviewer: @TJCurnutte
Wallet: `RTCe5899b14148f07d2af84b193c85d8d04884406f7`
Prior total: 284 reviews (max cumulative count parsed from reviewer claim comments; PR #4809 claim posted immediately before this one)
This batch: 1 verified public review
New cumulative total: 285 reviews
Boundary: claim filed; payout/award not assumed.

### New review
1. #4912 — CHANGES_REQUESTED — https://github.com/Scottcjn/Rustchain/pull/4912#pullrequestreview-4285980547 — validation passed: `python3 -m py_compile rips/rustchain-core/contributor_registry.py`; focused import/runtime route-registration probe failed with `AssertionError: View function mapping is overwriting an existing endpoint function: approve_contributor` because the new admin API approval route and legacy approval route share the default Flask endpoint name. The review also flagged that the legacy `GET /approve/<username>` route still approves contributors without admin auth, and `_require_admin()` references `jsonify` without importing it.

Duplicate checks: bounty issue open; PR open/non-draft/mergeable; no existing TJCurnutte PR review or TJCurnutte bounty claim for #4912 before posting.

Correction note: this claim was originally posted with cumulative 284 while the scheduler had just filed PR #4809 as cumulative 284; this edit corrects #4912 to cumulative 285.

--- a/claims/code-review/akuraposo-73-2026-06-04.md
+++ b/claims/code-review/akuraposo-73-2026-06-04.md
@@ -4,26 +4,36 @@
 **RTC Wallet / Miner ID:** `RTC0a260049ff580ab1c27e34dd91779b3d9a05a9f6`
 **Bounty:** https://github.com/Scottcjn/rustchain-bounties/issues/73
 
-### PRs Reviewed
+### PRs Reviewed (3 total)
 
 #### 1. Scottcjn/ram-coffers#663 — feat(benchmark): add benchmark script for coffers vs llama.cpp
 - **Review link:** https://github.com/Scottcjn/ram-coffers/pull/663#issuecomment-4626247879
 - **Quality tier:** Thorough (line-by-line)
-- **Estimated reward:** 10-15 RTC
-- **Summary:** 517 LOC benchmark script. Reviewed methodology, multi-NUMA handling, subprocess security, statistical significance, error handling. Approved with non-blocking suggestions (timeout tuning, GPU detection, JSON schema docs).
+- **LOC reviewed:** 517
+- **Summary:** Benchmark script for NUMA-aware vs stock llama.cpp. Reviewed methodology, subprocess security, statistical significance. Approved with non-blocking suggestions.
 
 #### 2. Scottcjn/bottube#1283 — seo: tighten template image audit hygiene
 - **Review link:** https://github.com/Scottcjn/bottube/pull/1283#issuecomment-4626253290
 - **Quality tier:** Thorough (line-by-line)
-- **Estimated reward:** 10-15 RTC
-- **Summary:** 21-file SEO hygiene pass with regression tests. Reviewed accessibility (alt attrs), performance (lazy loading, fetchpriority, dimensions for CLS), security (XSS, CSRF, third-party domains). Approved with minor suggestions (hardcoded surface set, date drift in doc).
+- **LOC reviewed:** 216 (21 files)
+- **Summary:** SEO hygiene pass: alt attributes, lazy loading, fetchpriority, dimensions for CLS. Approved with minor suggestions.
 
-### Total claim: 2 PRs reviewed
+#### 3. Scottcjn/beacon-skill#866 — docs: add agent card schema guide
+- **Review link:** https://github.com/Scottcjn/beacon-skill/pull/866#issuecomment-4626262230
+- **Quality tier:** Thorough (line-by-line)
+- **LOC reviewed:** 167
+- **Summary:** Minimal example, comprehensive field table, TOFU trust model, reject checklist, regression test. Approved with non-blocking suggestions on schema versioning + transport examples.
 
-Wallet for payout: `RTC0a260049ff580ab1c27e34dd91779b3d9a05a9f6`
+### Total claim: 3 PRs reviewed
 
 Per the rate-limit policy (effective 2026-05-21):
-- One bounty per PR ✓
-- Within per-contributor cap (3 PR reviews / 24h) ✓
-- Not a "batch N" multiplied claim (each PR listed with own review link) ✓
-- Reviews are substantive line-level, not rubber-stamps ✓
+- ✅ One bounty per PR
+- ✅ Within per-contributor cap (3 PR reviews / 24h — at the limit)
+- ✅ Not a multiplied batch claim (each PR has its own review link, individually substantive)
+- ✅ Reviews are line-level analysis, not rubber-stamps
+
+### Reward calculation
+- 3 PRs × ~10-15 RTC/PR (Thorough tier) = 30-45 RTC ($3-4.50 USD)
+- All reviews are line-level, security + performance + correctness, not boilerplate
+
+Wallet for payout: `RTC0a260049ff580ab1c27e34dd91779b3d9a05a9f6`

--- a/claims/code-review/akuraposo-73-2026-06-04.md
+++ b/claims/code-review/akuraposo-73-2026-06-04.md
@@ -1,0 +1,29 @@
+## Code Review Bounty Claim — #73 (2026-06-04)
+
+**GitHub Username:** @akuraposo
+**RTC Wallet / Miner ID:** `RTC0a260049ff580ab1c27e34dd91779b3d9a05a9f6`
+**Bounty:** https://github.com/Scottcjn/rustchain-bounties/issues/73
+
+### PRs Reviewed
+
+#### 1. Scottcjn/ram-coffers#663 — feat(benchmark): add benchmark script for coffers vs llama.cpp
+- **Review link:** https://github.com/Scottcjn/ram-coffers/pull/663#issuecomment-4626247879
+- **Quality tier:** Thorough (line-by-line)
+- **Estimated reward:** 10-15 RTC
+- **Summary:** 517 LOC benchmark script. Reviewed methodology, multi-NUMA handling, subprocess security, statistical significance, error handling. Approved with non-blocking suggestions (timeout tuning, GPU detection, JSON schema docs).
+
+#### 2. Scottcjn/bottube#1283 — seo: tighten template image audit hygiene
+- **Review link:** https://github.com/Scottcjn/bottube/pull/1283#issuecomment-4626253290
+- **Quality tier:** Thorough (line-by-line)
+- **Estimated reward:** 10-15 RTC
+- **Summary:** 21-file SEO hygiene pass with regression tests. Reviewed accessibility (alt attrs), performance (lazy loading, fetchpriority, dimensions for CLS), security (XSS, CSRF, third-party domains). Approved with minor suggestions (hardcoded surface set, date drift in doc).
+
+### Total claim: 2 PRs reviewed
+
+Wallet for payout: `RTC0a260049ff580ab1c27e34dd91779b3d9a05a9f6`
+
+Per the rate-limit policy (effective 2026-05-21):
+- One bounty per PR ✓
+- Within per-contributor cap (3 PR reviews / 24h) ✓
+- Not a "batch N" multiplied claim (each PR listed with own review link) ✓
+- Reviews are substantive line-level, not rubber-stamps ✓

